services/pipenv-helpers.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { InvalidParameter } = require('.')
+
+const isDependency = Joi.object({
+  version: Joi.string().required(),
+}).required()
+
+const isLockfile = Joi.object({
+  _meta: Joi.object({
+    requires: Joi.object({
+      python_version: Joi.string().required(),
+    }).required(),
+  }).required(),
+  default: Joi.object().pattern(Joi.string().required(), isDependency),
+  develop: Joi.object().pattern(Joi.string().required(), isDependency),
```

Comment:
Requiring _every_ dependency to have the version key will cause us to render `'invalid response data'` on lockfile that use any VCS dependencies. Example: https://github.com/DemocracyClub/aggregator-api/blob/5dc48baa0a45effadc238cf93954200bfdcc855b/Pipfile.lock#L134-L138

Reply:
Rather than try to handle everything I started with a naive implementation and figured we could improve it.

Thanks for the example. Will add a test case that covers this.

Reply:
It just happened the first repo I thought of to test with contains a VCS dependency, so I hit this issue immediately :)

> Is that reliably a VCS dependency?

What do you mean by that?

Reply:
> It just happened the first repo I thought of to test with contains a VCS dependency, so I hit this issue immediately :)

Nice 😁 

I'm thinking I'll add a service test that shows the version of a VCS dependency. Would that dependency on that repo be a good one to use?

Reply:
Have used that one for now.

Reply:
oh. I see what you mean. Yeah - its fine to point a live test at that repo